Move instruction works

olliethomas opened the pull request:
added martial arts and resting

olliethomas committed:
Move instruction works

added martial arts and resting

ElectricianX8 commented on AntGame/src/antgame/Match.java:
```diff
@@ -47,15 +59,12 @@ public Match(Player p1, Player p2, GameBoard b) {
 
         for (int i = 0; i < numOfRedAnts; i++) {
             ants[i].setCurrentPosition(redAntHillCoordinates.get(0));
-            board.getCellAtPosition(redAntHillCoordinates.get(0)).setAntColour(Colour.RED);
```
This line is needed to allow ant rendering in the game

ElectricianX8 commented on AntGame/src/antgame/Match.java:
```diff
@@ -47,15 +59,12 @@ public Match(Player p1, Player p2, GameBoard b) {
         }
         for (int i = numOfRedAnts; i < totalAnts; i++) {
             ants[i].setCurrentPosition(blackAntHillCoordinates.get(0));
-            board.getCellAtPosition(blackAntHillCoordinates.get(0)).setAntColour(Colour.BLACK);
```
This line is needed to allow ant rendering in the game

ElectricianX8 commented on AntGame/src/antgame/Match.java:
```diff
@@ -114,169 +114,253 @@ private void step(Ant ant) throws Exception {
+                        nextState = move.getStateToGoToIfBlocked();
+                    } else {
+                        currentCell.removeOccupation();
+                        cellToMoveTo.setOccupied(ant.getId());
```
Need 
  currentCell.removeAntColour(); //reset colour
  cellToMoveTo.setAntColour(ant.getColour()); //change colour

to remove previous colour and set next cell to allow ant rendering

olliethomas committed:
Match C&P'd from drive

olliethomas committed:
print board was throwing null pointer

olliethomas committed:
Think I've fixed the stopping bug

If ant was resting a next state wasn’t being set

olliethomas committed:
Save issue fixed.

Think i’ve sorted the save problem. It was adding a new line to the
file name.

Also did something to the tests a bit - nothing to report about

olliethomas committed:
removed incorrect testBrain

olliethomas committed:
AntBrain names now need to start with a '~'

Fixed the lever. Was a problem distinguishing between comments / file
names and Instructions. Ant Brain names now start with a ‘~’  to deal
with this. Also, the lever and parser were looking for ‘PickUp’, the
string should have been ‘Pickup’.  Also, tweaked the regex and improved
robustness so the lever can deal with Food food FOOD etc. Plan to do
the same with the parser,

olliethomas committed:
Flip RNG works to spec

At least i think so, tested on the first 25 outputs, not the full 99
yet. Not entirely sure why you need a bit mask rather than just the
modulo operator, but hey, it works and if anybody understands the
difference and can let me know that’d be great

olliethomas committed:
Changed the flip RNG to keep list size small
